pkg/controller/controller.go

```diff
@@ -97,16 +97,40 @@ func (c *Controller) initOperatorConfig() {
 		c.logger.Infoln("no ConfigMap specified. Loading default values")
 	}
 
-	// env var takes priority over the same param from the operator ConfigMap
-	watchedNamespace := os.Getenv("WATCHED_NAMESPACE")
-	if watchedNamespace != "" {
-		c.logger.Infof("Watch the %q namespace specified in the env variable WATCHED_NAMESPACE\n", watchedNamespace)
-		configMapData["watched_namespace"] = watchedNamespace
-	}
+	watchedNsConfigMapVar, isPresentInOperatorConfigMap := configMapData["watched_namespace"]
+	watchedNsEnvVar, isPresentInOperatorEnv := os.LookupEnv("WATCHED_NAMESPACE")
+
+	if (!isPresentInOperatorConfigMap) && (!isPresentInOperatorEnv) {
 
-	if configMapData["watched_namespace"] == "" {
 		c.logger.Infof("No namespace to watch specified. By convention, the operator falls back to watching the namespace it is deployed to: '%v' \n", spec.GetOperatorNamespace())
 		configMapData["watched_namespace"] = spec.GetOperatorNamespace()
+
+	}
+
+	if (isPresentInOperatorConfigMap) && (!isPresentInOperatorEnv) {
```

Comment:
Looks like there is no reason for this block of code here. It should be something like:


```
configmapData['watched_namespace'] = util.Coalesce(namespaceFromEnvironment, util.Coalesce(namespaceFromConfigMap, spec.GetOperatorNamespace()))
if configmapData['watched_namespace']  == "*" {
    configmapData['watched_namespace'] = v1.NamespaceAll
    c.logger.Infof("Listening to all namespaces")
} else {
    c.logger.Infof("Listenting to the specific namespace %s",  configmapData['watched_namespace'] )
}

```

Perhaps it makes sense to wrap the code above into a function like `getEffectiveNamespace(environmentNamespace, configNamespace string)` that returns the result assigned to the configMapData and also emits info messages.

We can also add the Coalesce function to the util package,
```
func Coalesce(val, default string) {
     if val == "" {
         return default
    }
    return val
}
```